### PR TITLE
Fix parsecsv no entries

### PR DESCRIPTION
**Pull request by jochman**
## Status
Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/17898

## Description
When parsing one lined CSV file - no entries returned (no headers). it fixed.
added unit tests for the bug above + extracted from test playbook\test playbook deleted.

## Screenshots
image

## Does it break backward compatibility?
   - No

## Must have
- [X] Tests
- [ ] Documentation (with link to it)
- [ ] Code Review

### CODE REVIEW

**Comment by jochman** [comment]
@kirbles19 please review the changelog https://github.com/demisto/content/pull/4090/files#diff-fe5d4df708dae663f12638c26eae51be

File: Scripts/ParseCSV/ParseCSV.py [hunk 1]
```diff
@@ -44,6 +44,7 @@ def unicode_dict_reader(csv_data, **kwargs):
     csv_reader = csv.DictReader((line.replace('\0', '') for line in csv_data), **kwargs)
     arr = []
     no_name_columns_counter = 0
+    # if csv_reader.
```
Comment (Itay4) [inline Scripts/ParseCSV/ParseCSV.py]: remove

File: Scripts/ParseCSV/ParseCSV.py [hunk 2]
```diff
@@ -84,162 +85,181 @@ def unicode_dict_reader(csv_data, **kwargs):
...
-
-demisto.results({
-    "Type": entryTypes["note"],
-    "ContentsFormat": formats["text"],
-    "Contents": content,
-    "HumanReadable": md,
-    "EntryContext": context
-})
+        entry = all_csv[0]
+        if not isinstance(entry, dict):
```
Comment (Itay4) [inline Scripts/ParseCSV/ParseCSV.py]: add comment

File: Scripts/ParseCSV/ParseCSV.py [hunk 3]
```diff
@@ -84,162 +85,181 @@ def unicode_dict_reader(csv_data, **kwargs):
...
+        demisto.results({
+            "Type": entryTypes["note"],
+            "ContentsFormat": formats["json"],
+            "ReadableContentsFormat": formats["markdown"],
+            "Contents": all_csv,
+            "EntryContext": output,
+            "HumanReadable": human_readable
+        })
+
+    elif not (parse_ip == -1 and parse_domain == -1 and parse_hash == -1):
```
Comment (Itay4) [inline Scripts/ParseCSV/ParseCSV.py]: handle one dimension here too

File: Scripts/ParseCSV/ParseCSV_test.py [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+        if file_obj:
+            TestParseCSV.mock_file_path(mocker, **file_obj)
+
+    @staticmethod
+    def get_demisto_results():
+        return demisto.results.call_args[0][0]
+
+    @staticmethod
+    def create_file_object(file_path):
+        # (str) -> dict
```
Comment (Itay4) [inline Scripts/ParseCSV/ParseCSV_test.py]: remove

**Comment by kirbles19** [comment]
@jochman Reviewed

**Comment by jochman** [comment]
> looks good
> 
> 1. run test playbook manually
> 2. add empty file to test
> 3. add file with non ascii chars

did it, all is good.

**Pull request merged**